.ci/integration-tests

```diff
@@ -183,58 +235,158 @@ function tc_machine_deployment() {
     printf "\nStarting TestCase: %s" "${FUNCNAME[0]}"
 
     # Wait until 3 nodes have joined the cluster
-    hf_object_create aws/md.yaml
-    printf "\n\tWaiting 600s for 3 machines to join the cluster"
-    hf_wait_on "hf_num_of_ready_nodes" nodes 3 600
+    hf_object_create $provider/md.yaml
+    printf "\n\tWaiting 1200s for 3 machines to join the cluster"
+    hf_wait_on "hf_num_of_ready_nodes" nodes 3 1200
 
     # Scale down the number of nodes to 2
-    hf_object_configure aws/md-scale-down.yaml
-    printf "\n\tWaiting 300s for machines to scale-down to 2"
-    hf_wait_on "hf_num_of_ready_nodes" nodes 2 300
+    hf_object_configure $provider/md-scale-down.yaml
+    printf "\n\tWaiting 1200s for machines to scale-down to 2"
+    hf_wait_on "hf_num_of_ready_nodes" nodes 2 1200
 
     # Update the machines to v2 and double the number of replicas
-    hf_object_configure aws/md-update.yaml
-    printf "\n\tWaiting 1200s for machines to upgrade to x.Large type and scale-up replicas to 4"
-    hf_wait_on "hf_num_of_ready_nodes" nodes 4 1200
+    hf_object_configure $provider/md-update.yaml
+    printf "\n\tWaiting 2400s for machines to upgrade to larger machine types and scale-up replicas to 4"
+    hf_wait_on "hf_num_of_ready_nodes" nodes 4 2400
 
     # Delete the machine-deployment
-    hf_object_delete aws/md.yaml
-    hf_wait_on "hf_num_of_objects" machdeploy 0 300
+    hf_object_delete $provider/md.yaml
+    printf "\n\tWaiting 1200s for machine to be deleted"
+    hf_wait_on "hf_num_of_objects" machdeploy 0 1200
 
     printf "\nCompleted TestCase\n"
 }
 
 ############################################## </TestCases> ##################################################
 ############################################## <CleanUp> #####################################################
 
+function clean_up() {
+    provider=$1
+    export KUBECONFIG=dev/control_kubeconfig_${provider}.yaml:dev/target_kubeconfig_${provider}.yaml
+
+    printf "\n\t\t\t----- CleanUp Test Environment -----\n"
+    hf_scale_deploy machine-controller-manager 1
+
+    if
+        hf_num_of_objects machdeploy 0
+        ret=$?
+        [[ $ret -ne 0 ]]
+    then
+        # Delete the machine-deployment
+        hf_object_delete $provider/md.yaml
+        printf "\n\tWaiting 1200s for machine-deployment to be deleted"
+        hf_wait_on "hf_num_of_objects" machdeploy 0 1200 $5
+    fi
+
+    if
```

Comment:
Should we add MachineSet as well?

Reply:
Added machine-set in the clean-up process as suggested.